lgc/patch/llpcPatchDescriptorLoad.cpp

```diff
@@ -297,9 +320,12 @@ Value *PatchDescriptorLoad::getDescPtr(ResourceNodeType resType, unsigned descSe
       offsetInDwords += DescriptorSizeResource / 4;
     offset = builder.getInt32(offsetInDwords);
   }
-  descPtr = builder.CreateBitCast(descPtr, builder.getInt32Ty()->getPointerTo(ADDR_SPACE_CONST));
-  descPtr = builder.CreateGEP(builder.getInt32Ty(), descPtr, offset);
-
+  // The LLVM's internal handling of GEP instruction results in a lot of junk code and prevented selection
```

Comment:
Re Jax's report that this change causes test agents to hang: I'm a bit suspicious of this edit here, if only because nothing else looks like it should affect normal pipeline compilation. You could try making it conditional on !isa<ConstantInt>(descPtr), falling back to the old GEP code when it is a ConstantInt, which it will always be for pipeline compilation.

Reply:
You are right. This is actually wrong when relocation is not used. In normal mode, offset is in dwords not bytes, so it cannot be just an add. I have corrected this issue and trying to test this before making a push. Thanks so much for pointing this out.

Reply:
Ah right, that would explain the crash. :-)

I don't like the offset being in dwords for one path and bytes for the other path. I'll have a look at changing the normal path to bytes in a separate change.